addons/l10n_it_edi/models/account_edi_format.py

```diff
@@ -433,18 +429,23 @@ def _import_fattura_pa(self, tree, invoice):
                                             invoice_line_form.product_id = product
                                             break
                                     if partner:
-                                        product_supplier = self.env['product.supplierinfo'].search([('name', '=', partner.id), ('product_code', '=', code.text)], limit=1)
-                                        if product_supplier and product_supplier.product_id:
+                                        product_supplier = self.env['product.supplierinfo'].search([('name', '=', partner.id), ('product_code', '=', code.text)], limit=2)
+                                        if product_supplier and len(product_supplier) == 1 and product_supplier.product_id:
                                             invoice_line_form.product_id = product_supplier.product_id
                                             break
                                 if not invoice_line_form.product_id:
                                     for element_code in elements_code:
                                         code = element_code.xpath('.//CodiceValore')[0]
-                                        product = self.env['product.product'].search([('default_code', '=', code.text)], limit=1)
-                                        if product:
+                                        product = self.env['product.product'].search([('default_code', '=', code.text)], limit=2)
+                                        if product and len(product) == 1:
                                             invoice_line_form.product_id = product
                                             break
 
+                            # Label.
+                            line_elements = element.xpath('.//Descrizione')
+                            if line_elements:
+                                invoice_line_form.name = " ".join(line_elements[0].text.split())
```

Comment:
@dacd-odoo I wonder what the use could be for this split, join

Reply:
It's used to remove whitespaces (\t \n ...) into simple space